### ISSUE TITLE
Consistent string formatting, esp. update long string creation via `+=`

### ISSUE THREAD

**Issue by sadielbartholomew** [NCAS-CMS/cf-plot#46]
After #45, where auto-formatting was applied to abide by PEP8, I notice we are still left with numerous cases of strings, often longer and/or inclusive of variable string-representation reporting, which are formatted in a mixture of ways:

https://github.com/NCAS-CMS/cf-plot/blob/7164605327735931951719bd3d3bef5d804a557f/cfplot/cfplot.py#L742-L745
and/or using variable iterative appending, e.g:
https://github.com/NCAS-CMS/cf-plot/blob/7164605327735931951719bd3d3bef5d804a557f/cfplot/cfplot.py#L1642-L1655

which can be replaced by wrapping in parentheses and stating each component on a new line, where they will automatically be merged into one.

We should convert these so they use a consistent formatting, for all but cases where there is a clear reason to use a different approach e.g. clarity when `{` is needed in the string. Since it is used throughout cf-python and is what  am most familiar with, I will use f-string formatting, where we can prefer `.format` as a second choice if there's a reason f-string aren't clear for a given case. We should therefore remove all cases of old-style/`%` formatting.

**Comment by davidhassell** [comment]
+1 all round on this. When it comes it `{` inside f-strings, I don't mind the triple brackets thing, but I think that they're perhaps pretty rare, anyway.

**Comment by sadielbartholomew** [comment]
Thanks David. Strongly suspected you'd be happy with this change!

> When it comes it { inside f-strings, I don't mind the triple brackets thing, but I think that they're perhaps pretty rare, anyway.

I highlight that case mostly because, unlike for the general case where it is rare as you point out, it isn't so rare in cf-plot, as I've seen, since [matplotlib supports (La)TeX formatting for text](https://matplotlib.org/stable/gallery/text_labels_and_annotations/tex_demo.html#rendering-math-equations-using-tex) e.g. titles and axes labels, so there is quite a lot of processing of TeX via `{, }` characters in the code, e.g:

https://github.com/NCAS-CMS/cf-plot/blob/7164605327735931951719bd3d3bef5d804a557f/cfplot/cfplot.py#L4347-L4352

Hence why I think it is wise to agree a strategy for those. In which case, I'm not against triple brackets by using f-strings, but I feel it could lead to some bugs especially when we need to nest braces for the TeX, so maybe `.format` is better in those cases. What do you think?
